File: template.js
var fs = require('fs');

var isDebugEnabled = true;
var isReadFromFile = true;
var dataFile = "template.txt";

function readFromFile() {
  var text = fs.readFileSync(dataFile, 'utf-8');
  var lines = text.split('\n').map(x => x.trim()).filter(x => x.length > 0);
  return lines;
}

function readFromStdin(exec) {
  process.stdin.setEncoding('utf8');
  var reader = require('readline').createInterface({
      input: process.stdin,
      output: process.stdout
  });

  var lines = [];
  reader.on('line', function (line) {
    line = line.trim();
    if (line.length > 0) {
      lines.push(line);
    }
  });

  reader.on('close', () => {
    exec(lines);
  });
}

function strListToIntList(list) {
  return list.map(x1 => x1.split(' ').map(x2 => Number(x2)))
}

// ファイルと標準入力のきりわけ
function main() {
  if (isReadFromFile) {
    var lines = readFromFile();
    exec(lines);
  } else {
    readFromStdin(exec);
  }
}

function debug(str) {
  if (!isDebugEnabled) {
    return;
  }

  console.log(str)
}

// メイン処理
function exec(lines) {
  debug(lines)

  var input_nums = strListToIntList(lines);
  debug(input_nums)
}

main();

File: template.py
from sys import stdin
from logging import getLogger, StreamHandler, DEBUG

is_read_from_file = True
is_debug_enabled = True
data_file = "template.txt"

log = getLogger(__name__)
handler = StreamHandler()
handler.setLevel(DEBUG)
log.setLevel(DEBUG)
log.addHandler(handler)
log.propagate = False
log.disabled = (not is_debug_enabled)


def read_input_from_stdin():
    input_lines = filter(lambda l: len(l), map(lambda l2: l2.strip(), stdin.readlines()))

    return input_lines


def read_input_from_file():
    with open(data_file, mode='r') as f:
        input_lines = list(filter(lambda row: row, f.read().split('\n')))
    return input_lines


def str_list_to_int_list(a):
    return list(map(lambda x: list(map(int, x.split(' '))), a))


def str_list_to_list_of_str_list(a):
    return list(map(lambda x: x.split(' '), a))


def main():
    if is_read_from_file:
        input_lines = read_input_from_file()
    else:
        input_lines = read_input_from_stdin()

    log.debug('input_lines: %s', input_lines)

    input_values = str_list_to_int_list(input_lines)

    log.debug('input_values: %s', input_values)


if __name__ == '__main__':
    main()
